File: BurgerBar/ClientApp/src/app/models/burger.ts
import { Bun } from './bun';
import { Ingredient } from './ingredient';
import { Product } from './product';

export class Burger {
    id?: number;
    name: string;
    price: number;
    code: string;
    bun: Bun;
    ingredients: Array<Ingredient> = [];
}

File: BurgerBar/Entities/Burger.cs
using BurgerBar.Resources.Localization;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace BurgerBar.Entities
{
    public class Burger : Product
    {
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public string Code { get; set; }
        
        public Bun Bun { get; set; }

        [Required(ErrorMessageResourceName = "IngredientsRequiredError", ErrorMessageResourceType = typeof(ErrorMessages))]
        [MinLength(1, ErrorMessageResourceName = "IngredientsRequiredError", ErrorMessageResourceType = typeof(ErrorMessages))]
        public ICollection<BurgerIngredient> Ingredients { get; set; }

        public CreationType CreationType { get; set; }
    }
}
